src/wasi/Wasi.cpp

```diff
@@ -64,10 +64,13 @@ void WASI::proc_exit(ExecutionState& state, Value* argv, Value* result, Instance
     ASSERT_NOT_REACHED();
 }
 
+void WASI::proc_raise(ExecutionState& state, Value* argv, Value* result, Instance* instance)
+{
+    result[0] = Value(uvwasi_proc_raise(WASI::m_uvwasi, argv[0].asI32()));
+}
+
 void WASI::random_get(ExecutionState& state, Value* argv, Value* result, Instance* instance)
 {
-    ASSERT(argv[0].type() == Value::I32);
-    ASSERT(argv[1].type() == Value::I32);
```

Comment:
IMHO these `ASSERT` are useful for debugging because we can easily check the reason of fail in debug mode.
(`ASSERT` is enabled only for debug mode build)
Other WASI function such as `proc_exit` also checks the type of argument using `ASSERT`.

Reply:
A agree with you about their importancy. But this is checked in the function when their value is used. For example, by calling `argv[0].asI32()`, it checks its type.
```C
int32_t Value::asI32() const {
    ASSERT(type() == I32);
    return m_i32;
}
```
In light of this information, shall I add the asserts?
(and sorry for not replying for so long)

Reply:
I think that these ASSERT are more intuitive for checking errors.
Also, other functions like `WASI::proc_exit` checks the type before the value is used.